File: Patterns/9. Diamond Patterns/89. Full Diamond Repeated Row Numbers/fdrrn.cpp
#include <iostream>
using namespace std;
int main(){
    int n,row;
    cout<<"Enter a Number : ";
    cin>>n;

    for(int i=1; i<=2*n-1; i++){
        if(i<=n){
            row = i;
        }else{
            row = 2*n-i;
        }
        // print spaces
        for(int j=1; j<=n-row; j++){
            cout<<"  ";
        }
        // print first half
        for(int j=1; j<=row; j++){
            cout<<i<<" ";
        }
        // print second half
        for(int j=row-1; j>=1; j--){
            cout<<i<<" ";
        }
        cout<<'\n';

    }
}

/*
Output

        1 
      2 2 2 
    3 3 3 3 3 
  4 4 4 4 4 4 4 
5 5 5 5 5 5 5 5 5 
  6 6 6 6 6 6 6 
    7 7 7 7 7 
      8 8 8 
        9 

*/

File: Patterns/9. Diamond Patterns/89. Full Diamond Repeated Row Numbers/fdrrn.py
n = int(input("Enter a Number : "))
row=1

for i in range(1,2*n):
    if(i<=n):
        row=i
    else:
        row=2*n-i
    
    # print spaces
    for j in range(1,n-row+1):
        print(end='  ')
    # print first half
    for j in range(1,row+1):
        print(i, end=' ')
    # print second half
    for j in range(row-1,0,-1):
        print(i, end=' ')
    print()

'''
Output:

        1 
      2 2 2 
    3 3 3 3 3 
  4 4 4 4 4 4 4 
5 5 5 5 5 5 5 5 5 
  6 6 6 6 6 6 6 
    7 7 7 7 7 
      8 8 8 
        9
         
'''